spec/comment_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'creation' do
+    before do
+      @user = User.create(name: 'John Doe', photo: 'https://picsum.photos/200/300', bio: 'I am John Doe')
+      @post = Post.create(author_id: @user.id, title: 'My Post', text: 'This is my post')
+      @comment = Comment.create(author_id: @user.id, post_id: @post.id, text: 'This is my comment')
+    end
+
+    it "text can't be blank" do
+      @comment.text = nil
+      expect(@comment).to_not be_valid
+    end
+
+    it "author_id can't be blank" do
+      @comment.author_id = nil
+      expect(@comment).to_not be_valid
+    end
+
+    it 'author_id must be an integer' do
+      @comment.author_id = 'a'
+      expect(@comment).to_not be_valid
+    end
+
+    it 'author_id must be greater than 0' do
+      expect(@comment).to_not be_valid
+    end
+
+    it "post_id can't be blank" do
+      @comment.post_id = nil
+      expect(@comment).to_not be_valid
+    end
+
+    it 'post_id must be an integer' do
+      @comment.post_id = 'a'
+      expect(@comment).to_not be_valid
+    end
+
+    it 'post_id must be greater than 0' do
+      expect(@comment).to_not be_valid
+    end
+  end
+end
```

Comment:
- Two test blocks in this spec file are failing, and this is because the tests are actually valid, though you expect them to be invalid. Please note that the `before` method uses the `before :each` by default, so it runs the code in the before block before every test, even though you might have set the `author_id or post_id` to nil in previous tests. 😄 
- Alternatively, you could use the `before :all` method so the before block is run only once before all your tests 😄

Reply:
Fixed.